src/Adapter/MSTest.TestAdapter/Helpers/ReflectHelper.cs

```diff
@@ -278,11 +278,22 @@ internal static TestIdGenerationStrategy GetTestIdGenerationStrategy(Assembly as
     /// Gets TestDataSourceDiscovery assembly level attribute.
     /// </summary>
     /// <param name="assembly"> The test assembly. </param>
+    [Obsolete]
```

Comment:
As it's internal, why not remove instead of obsoleting?
And if obsoleting is necessary, I think it may be good to add a message "Use 'GetTestDataSourceOptions' instead".

Reply:
Type is obsolete so it forces me to suppress the obsolete. It feels easiers and clearer to have the method also obsolete. It will also help with cleanup IMO when we are finally getting to v4.